Build for sqlite broken: File exists: '/home/travis/.python-eggs'

Sqlite tests are sporadically breaking in unrelated requests like #574 and maybe #573, though not always (since the tests for #573 look clean until after it was merged).

The fact that the main error is `E     [Errno 17] File exists: '/home/travis/.python-eggs'` makes me think this is being caused by the test parallelization I had suggested for only sqlite in #561.

A short-term fix might be to disable parallel tests, though ideally in the long term we should fix that.

@knikolla - would you mind taking a look at this? @SahilTikale  - any thoughts?

The error is ([link to full output](https://travis-ci.org/CCI-MOC/haas/jobs/122858845)):

```
________________ ERROR collecting tests/unit/class_resolver.py _________________
tests/unit/class_resolver.py:3: in <module>
    from haas.model import *
../../../virtualenv/python2.7.9/lib/python2.7/site-packages/_pytest/assertion/rewrite.py:159: in load_module
    py.builtin.exec_(co, mod.__dict__)
haas/model.py:27: in <module>
    from flask.ext.sqlalchemy import SQLAlchemy
../../../virtualenv/python2.7.9/lib/python2.7/site-packages/Flask-0.10.1-py2.7.egg/flask/exthook.py:62: in load_module
    __import__(realname)

[...]
    self._extract_resource(manager, self._eager_to_zip(name))
../../../virtualenv/python2.7.9/lib/python2.7/site-packages/pkg_resources/__init__.py:1853: in _extract_resource
    self.egg_name, self._parts(zip_path)
../../../virtualenv/python2.7.9/lib/python2.7/site-packages/pkg_resources/__init__.py:1193: in get_cache_path
    self.extraction_error()
../../../virtualenv/python2.7.9/lib/python2.7/site-packages/pkg_resources/__init__.py:1173: in extraction_error
    raise err

E   ExtractionError: Can't extract file(s) to egg cache
E   
E   The following error occurred while trying to extract file(s) to the Python egg
E   cache:
E   
E     [Errno 17] File exists: '/home/travis/.python-eggs'
E   
E   The Python egg cache directory is currently set to:
E   
E     /home/travis/.python-eggs
E   
E   Perhaps your account does not have write access to this directory?  You can
E   change the cache directory by setting the PYTHON_EGG_CACHE environment
E   variable to point to an accessible directory.
```

--- a/ci/apache/install.sh
+++ b/ci/apache/install.sh
@@ -16,6 +16,7 @@ fi
 # Address #577 via
 # https://stackoverflow.com/questions/2192323/what-is-the-python-egg-cache-python-egg-cache
 mkdir -p ~/.python-eggs
+chmod go-w ~/.python-eggs # Eliminate "writable by group/others" warnings
 
 # Install HIL
 python setup.py install

--- a/ci/apache/install.sh
+++ b/ci/apache/install.sh
@@ -13,6 +13,10 @@ if [ $DB = postgres ]; then
     psql -c 'CREATE DATABASE hil;' -U postgres
 fi
 
+# Address #577 via
+# https://stackoverflow.com/questions/2192323/what-is-the-python-egg-cache-python-egg-cache
+mkdir -p ~/.python-eggs
+
 # Install HIL
 python setup.py install
 

--- a/ci/run_unit_tests.sh
+++ b/ci/run_unit_tests.sh
@@ -1,9 +1,9 @@
 #!/usr/bin/env bash
-# On SQLite we run the tests in parallel. This speeds things up
-# substantially, but is currently only safe for sqlite, since it
-# uses an in memory (and  therefore per-process) database.
 
-# Temporarily disabled till we figure out how to solve https://github.com/CCI-MOC/hil/issues/577
+# On SQLite we run the tests in parallel.
+# This speeds things up, but is currently only safe for sqlite, since it uses
+# an in memory (and  therefore per-process) database.
+
 if [ $DB = sqlite ]; then
     extra_flags='-n auto'
 fi

--- a/ci/run_unit_tests.sh
+++ b/ci/run_unit_tests.sh
@@ -4,9 +4,9 @@
 # uses an in memory (and  therefore per-process) database.
 
 # Temporarily disabled till we figure out how to solve https://github.com/CCI-MOC/hil/issues/577
-# if [ $DB = sqlite ]; then
-#     extra_flags='-n auto'
-# fi
+if [ $DB = sqlite ]; then
+    extra_flags='-n auto'
+fi
 
 py.test $extra_flags \
 	tests/custom_lint.py \

